### PR TITLE
CAD-391 eliding messages

### PR DESCRIPTION
**Pull request by CodiePP**
prints first and last message of range of equivalent 'TraceLedgerReplayEvent'.
```
[nixos19:cardano.node.ChainDB:Info:64] [2019-12-24 23:26:57.00 UTC] {"event":{"kind":"TraceOpenEvent.OpenedVolDB"},"kind":"TraceEvent","tip":"genesis(origin)"}
[nixos19:cardano.node.ChainDB:Info:64] [2019-12-24 23:26:57.31 UTC] {"event":{"snapshot":{"snapshot":"DiskSnapshot 9","kind":"snapshot"},"kind":"TraceLedgerReplayEvent.ReplayFromSnapshot","tip":"Tip (At (Block {blockPointSlot = SlotNo {unSlotNo = 3549092}, blockPointHash = ByronHash {unByronHash = AbstractHash 2cfa35fe9617a9c41ef5ec19bed50742e587ab10d18ce6ac80c09bf71308fc3e}}))"},"kind":"TraceEvent","tip":"genesis(origin)"}
[nixos19:cardano.node.ChainDB:Info:64] [2019-12-24 23:26:57.39 UTC] {"event":{"snapshot":{"snapshot":"DiskSnapshot 9","kind":"snapshot"},"kind":"TraceLedgerReplayEvent.ReplayFromSnapshot","tip":"Tip (At (Block {blockPointSlot = SlotNo {unSlotNo = 3549092}, blockPointHash = ByronHash {unByronHash = AbstractHash 2cfa35fe9617a9c41ef5ec19bed50742e587ab10d18ce6ac80c09bf71308fc3e}}))"},"kind":"TraceEvent","tip":"genesis(origin)"}
[nixos19:cardano.node.ChainDB:Info:64] [2019-12-24 23:26:57.39 UTC] messages elided in total = 70
[nixos19:cardano.node.ChainDB:Info:64] [2019-12-24 23:26:57.39 UTC] {"event":{"kind":"TraceOpenEvent.OpenedLgrDB"},"kind":"TraceEvent","tip":"genesis(origin)"}
```

### CODE REVIEW

**Comment by CodiePP** [comment]
depends on PR https://github.com/input-output-hk/iohk-monitoring-framework/pull/495

**Comment by CodiePP** [comment]
eliding `TraceLedgerReplayEvent`:
```
[nixos19:cardano.node.ChainDB:Info:63] [2019-12-30 13:06:34.52 UTC] {"event":{"snapshot":{"snapshot":"DiskSnapshot 19","kind":"snapshot"},"kind":
"TraceLedgerReplayEvent.ReplayFromSnapshot","tip":"Tip (At (Block {blockPointSlot = SlotNo {unSlotNo = 3560492}, blockPointHash = ByronHash {unBy
ronHash = AbstractHash 3ac09146dbc3c75e298543a22223ac3aeeab387d9ddecedac5cbe6465db34dc0}}))"},"kind":"TraceEvent","tip":"genesis(origin)"}
[nixos19:cardano.node.ChainDB:Info:63] [2019-12-30 13:06:34.78 UTC] before next, messages elided = 1432
[nixos19:cardano.node.ChainDB:Info:63] [2019-12-30 13:06:34.78 UTC] {"event":{"kind":"TraceLedgerReplayEvent.ReplayedBlock","slot":"3542400","epoch":"164"},"kind":"TraceEvent","tip":"genesis(origin)"}
```

eliding `TraceGCEvent`:
```
[nixos19:cardano.node.ChainDB:Debug:79] [2019-12-30 13:07:09.88 UTC] {"event":{"kind":"TraceGCEvent.PerformedGC","slot":{"kind":"SlotNo","slot":3562021}},"kind":"TraceEvent","tip":"4cac1a8@3567746"}
[nixos19:cardano.node.ChainDB:Debug:81] [2019-12-30 13:07:09.88 UTC] before next, messages elided = 2
[nixos19:cardano.node.ChainDB:Debug:81] [2019-12-30 13:07:09.88 UTC] {"event":{"kind":"TraceGCEvent.PerformedGC","slot":{"kind":"SlotNo","slot":3562025}},"kind":"TraceEvent","tip":"4cac1a8@3567746"}
```

**Comment by CodiePP** [comment]
> depends on PR [input-output-hk/iohk-monitoring-framework#495](https://github.com/input-output-hk/iohk-monitoring-framework/pull/495)

upstream PR has been merged. now ready.

File: cardano-node/src/Cardano/Tracing/Tracers.hs [hunk 1]
```diff
@@ -118,6 +119,26 @@ nullTracers = Tracers {
       muxTracer = nullTracer
     }
 
+
+instance ElidingTracer
+  (WithSeverity (WithTip blk (ChainDB.TraceEvent blk))) where
+  -- equivalent by type and severity
+  isEquivalent (WithSeverity s1 (WithTip _tip1 (ChainDB.TraceLedgerReplayEvent _ev1)))
```
Comment (ksaric) [inline cardano-node/src/Cardano/Tracing/Tracers.hs]: If they are both `TraceLedgerReplayEvent` and have the same severity? No need to match on the actual event of `TraceLedgerReplayEvent`?

File: benchmarking/chain-sync/benchmark-chain-sync-mainnet.sh [hunk 2]
```diff
@@ -9,18 +9,19 @@ DATADIR=state-node-mainnet
 mkdir -p $DATADIR
 cd $DATADIR
 
-if [ -d db-mainnet-0 ]; then
-  rm -rf db-mainnet-0
-fi
-rm node-0*
+#if [ -d db-mainnet-0 ]; then
```
[the remark this answers is not in the document]
Comment (CodiePP) [reply benchmarking/chain-sync/benchmark-chain-sync-mainnet.sh]: hm, no, this is from debugging

File: benchmarking/chain-sync/benchmark-chain-sync-mainnet.sh [hunk 3]
```diff
@@ -9,18 +9,19 @@ DATADIR=state-node-mainnet
...
 
 #set -euo pipefail
 
-ulimit -t $CPU_TIME_LIMIT
+#ulimit -t $CPU_TIME_LIMIT
 
 date --iso-8601=seconds > STARTTIME
 
 NODE="cabal v2-run exe:cardano-node -- "
+NODE="stack --nix exec cardano-node -- "
```
[the remark this answers is not in the document]
Comment (CodiePP) [reply benchmarking/chain-sync/benchmark-chain-sync-mainnet.sh]: yes, switched to 'stack'. Changing the order of the definitions will use stack/cabal.

File: benchmarking/chain-sync/configuration/log-configuration.yaml [hunk 4]
```diff
@@ -68,6 +68,8 @@ options:
       # Change the `subtrace` value to `Neutral` in order to log
       # `UTxO`-related messages during epoch validation.
       subtrace: NoTrace
+    'cardano.#messagecounters':
```
[the remark this answers is not in the document]
Comment (CodiePP) [reply benchmarking/chain-sync/configuration/log-configuration.yaml]: once that PR is merged, then no more.

File: cardano-node/src/Cardano/Tracing/Tracers.hs (hunk 1 above)
Comment (CodiePP) [reply cardano-node/src/Cardano/Tracing/Tracers.hs]: equivalence just by comparing severity and the type; could be made more complicated..

**Pull request merged**